kubeflow/trainer/types/types.py

```diff
@@ -41,7 +43,7 @@ class CustomTrainer:
     func: Callable
     func_args: Optional[Dict] = None
     packages_to_install: Optional[list[str]] = None
-    pip_index_url: str = constants.DEFAULT_PIP_INDEX_URL
+    pip_index_urls: list[str] = field(default_factory=lambda: constants.DEFAULT_PIP_INDEX_URLS)
```

Comment:
Let's update this as discussed in https://github.com/kubeflow/sdk/pull/74#discussion_r2310104972
```suggestion
    pip_index_urls: list[str] = field(default_factory=lambda: list(constants.DEFAULT_PIP_INDEX_URLS))
```